c/src/ml-api-common.c

```diff
@@ -26,7 +26,7 @@ ml_tensors_info_create (ml_tensors_info_h * info)
 {
   ml_tensors_info_s *tensors_info;
 
-  check_feature_state ();
+  check_feature_state (ML_INF_FEATURE);
```

Comment:
Need discussion: This is common util function. It needs to check all feature - inference / training / service. (Or, machine-learning?)

Reply:
Tensor utility functions such as ml_tensors_info_create() could be used with the NNTrainer module as below.
* https://github.com/nnstreamer/nntrainer/blob/main/api/capi/src/nntrainer.cpp#L1043-L1046
```cpp
  status = ml_tensors_info_create(info);
  if (status != ML_ERROR_NONE) {
    return status;
  }
```

NNTrainer API also checks the feature. However, the name of the macro `check_feature_state()` is exactly the same but NNTrainer defines its own macro instead of using a common function.
* https://github.com/nnstreamer/nntrainer/blob/main/api/capi/include/nntrainer_internal.h#L55-L61
```cpp
#define check_feature_state()                         \
  do {                                                \
    int feature_ret = ml_tizen_get_feature_enabled(); \
    if (ML_ERROR_NONE != feature_ret)                 \
      return feature_ret;                             \
  } while (0);
```

In my opinion, it is not a good idea since a single process has two different logic for checking the Tizen feature. I hope that `check_feature_state()` in ml-api-common.c will be provided as common utility function.

Reply:
Right. It can be commonly used in the NNTrainer.
Using a common function is more helpful for code management and debugging.
I will update the NNTrainer for feature check with these changes. Thanks.